go/private/rules/cgo.bzl

```diff
@@ -91,14 +91,28 @@ def _select_archive(files):
     for f in files:
       if f.basename.endswith(ext):
         return f
-  fail("cc_library did not produce any files")
+
+def _select_archives(libs):
+  """Selects a one per item in a cc_library, if needed.
+
+  If no archive can be extracted from all the libraries, this will fail.
+  """
+  # list of file extensions in descending order or preference.
+  outs = []
+  for lib in libs:
+    archive = _select_archive(lib.files)
+    if archive:
```

Comment:
When would a `cc_library` not produce any usable archives? Does that happen if there are no source files?

Reply:
This can happen if there are only header files (which happens if C++ files are missing, for instance).

Reply:
I realize I could have been clearer. In the case of a regular C compilation, the `cc_library` responsible of building the C++ files will not produce any archive.